sensing/autoware_pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp

```diff
@@ -187,15 +188,99 @@ bool DistortionCorrector<T>::isInputValid(sensor_msgs::msg::PointCloud2 & pointc
   return true;
 }
 
+template <class T>
+bool DistortionCorrector<T>::azimuthConversionExists(sensor_msgs::msg::PointCloud2 & pointcloud)
+{
+  if (!isInputValid(pointcloud)) return false;
+
+  sensor_msgs::PointCloud2Iterator<float> it_x(pointcloud, "x");
+  sensor_msgs::PointCloud2Iterator<float> it_y(pointcloud, "y");
+  sensor_msgs::PointCloud2Iterator<float> it_azimuth(pointcloud, "azimuth");
+
+  auto next_it_x = it_x;
+  auto next_it_y = it_y;
+  auto next_it_azimuth = it_azimuth;
+
+  if (it_x != it_x.end() && it_x + 1 != it_x.end()) {
+    next_it_x = it_x + 1;
+    next_it_y = it_y + 1;
+    next_it_azimuth = it_azimuth + 1;
+  } else {
+    RCLCPP_WARN(
+      node_->get_logger(),
+      "Current point cloud only has a single point. Could not calculate the formula.");
+    return false;
+  }
+
+  for (; next_it_x != it_x.end();
+       ++it_x, ++it_y, ++it_azimuth, ++next_it_x, ++next_it_y, ++next_it_azimuth) {
+    auto current_cartesian_rad = autoware::universe_utils::opencv_fastAtan2(*it_y, *it_x);
+    auto next_cartesian_rad = autoware::universe_utils::opencv_fastAtan2(*next_it_y, *next_it_x);
+
+    // If the angle exceeds 180 degrees, it may cross the 0-degree axis,
+    // which could disrupt the calculation of the formula.
+    if (
+      abs(*next_it_azimuth - *it_azimuth) >= autoware::universe_utils::pi ||
+      abs(next_cartesian_rad - current_cartesian_rad) >= autoware::universe_utils::pi) {
+      RCLCPP_WARN(
+        node_->get_logger(),
+        "Angle between two points exceeds 180 degrees. Iterate to next point ...");
+      continue;
+    }
+
+    float b = (*next_it_azimuth - *it_azimuth) / (next_cartesian_rad - current_cartesian_rad);
+    float a = *it_azimuth - b * current_cartesian_rad;
+
+    // Check if 'b' can be adjusted to 1 or -1
+    if (std::abs(b - 1.0f) <= threshold_b_) {
+      b_ = 1.0f;
+    } else if (std::abs(b + 1.0f) <= threshold_b_) {
+      b_ = -1.0f;
+    } else {
+      RCLCPP_WARN(
+        node_->get_logger(),
+        "Angle between two points exceeds 180 degrees. Iterate to next point ...");
+      std::cout << "b: " << b << "should be close to 1 or -1" << std::endl;
```

Comment:
Can you replace to rclcpp_foo?

Reply:
Thanks for the check! fixed in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/f509d7d84d5fa92aeb7e6357da3446c803383270